src/client/pages/AssessmentSection/DataTable/Table/utils/parseTable.ts

```diff
@@ -2,23 +2,29 @@ import { Arrays } from '@utils/arrays'
 import { UUIDs } from '@utils/uuids'
 
 import { CountryIso } from '@meta/area'
-import { Col as TypeCol, Cycle, Row as TypeRow, RowType, Table } from '@meta/assessment'
-import { TableData } from '@meta/data'
+import { AssessmentName, Col as TypeCol, Cycle, Row as TypeRow, RowType, Table } from '@meta/assessment'
+import { RecordAssessmentData } from '@meta/data'
 
 type Props = {
+  assessmentName: AssessmentName
   countryIso: CountryIso
   cycle: Cycle
-  data: TableData
+  data: RecordAssessmentData
   showODP: boolean
   table: Table
 }
 
 const getHeaders = (props: Props): string[] => {
-  const { countryIso, cycle, data, showODP, table } = props
+  const { assessmentName, countryIso, cycle, data, showODP, table } = props
 
   const headers = table.props.columnNames[cycle.uuid]
   if (table.props.odp && showODP) {
-    const headersDiff = data ? Arrays.difference(Object.keys(data[countryIso]?.[table.props.name] ?? {}), headers) : []
+    const headersDiff = data?.[assessmentName]?.[cycle.name]
+      ? Arrays.difference(
```

Comment:
better to add  a util method  to extract data?.[assessmentName]?.[cycle.name] and data?.[assessmentName]?.[cycle.name]?.[countryIso]?.[table.props.name] ? what do you think ?